src/ui/Pager/Pager.ts

```diff
@@ -146,6 +144,26 @@ export class Pager extends Component {
     element.appendChild(this.list);
   }
 
+  /**
+   * The current page (1-based index).
+   */
+  public get currentPage(): number {
+    return this._currentPage;
+  }
+
+  public set currentPage(value: number) {
+    let sanitizedValue = value;
+
+    if (isNaN(value)) {
+      this.logger.warn(`Unable to set pager current page to an invalid value: ${value}. Resetting to 1.`);
+      sanitizedValue = 1;
+    }
+
+    sanitizedValue = Math.round(sanitizedValue);
+
+    this._currentPage = sanitizedValue;
+  }
+
   /**
    * Sets the current page, then executes a query.
    *
```

Comment:
I cannot leave a comment but on line 177, we are using the `currentPage` setter as follows:

```
this.currentPage = Math.max(Math.min(pageNumber, this.getMaxNumberOfPagesForCurrentResultsPerPage()), 1);
```

We should move the math logic into the setter:
```
Math.max(Math.min(pageNumber, this.getMaxNumberOfPagesForCurrentResultsPerPage()), 1);
```

Otherwise, the person calling the setter would need to "know" what value to pass. This should instead be taken care of by the setter.